runtime/nvqir/custatevec/CuStateVecCircuitSimulator.cu

```diff
@@ -194,39 +261,99 @@ protected:
         controls32.size()));
   }
 
+  /// @brief Nice utility function to have to print the state vector contents on
+  /// GPU.
+  void printStateFromGPU(const std::string &name, void *ptr, std::size_t size) {
+    std::vector<std::complex<ScalarType>> tmp(size);
+    cudaMemcpy(tmp.data(), ptr, size * sizeof(std::complex<ScalarType>),
+               cudaMemcpyDeviceToHost);
+    for (auto &r : tmp)
+      printf("%s: %.12lf\n", name.c_str(), r.real());
```

Comment:
Should we also print the imaginary part?

Reply:
Yea definitely, will fix